libethereum/BlockChainSync.cpp

```diff
@@ -742,7 +742,7 @@ void BlockChainSync::resetSync()
 	m_bodySyncPeers.clear();
 	m_headerIdToNumber.clear();
 	m_syncingTotalDifficulty = 0;
-	m_state = SyncState::Idle;
+	m_state = SyncState::NotSynced;
```

Comment:
I'm not 100% sure it won't break anything but it doesn't look right at all.

Currently this `BlockChainSync::resetSync()` may be called during sync from https://github.com/ethereum/cpp-ethereum/blob/develop/libethereum/BlockChainSync.cpp#L601
Then `BlockChainSync::isSyncing()` reports that sync is complete and `Client` begins to import new transactions from TransactionQueue here: https://github.com/ethereum/cpp-ethereum/blob/develop/libethereum/Client.cpp#L676 
This shouldn't happen until sync is really complete.